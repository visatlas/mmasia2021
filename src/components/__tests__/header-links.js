import React from "react";
import renderer from "react-test-renderer";
import { NavLink, MobileNavLink } from "../header-links";

describe("Nav Link", () => {
  it("renders correctly", () => {
    const link = renderer
      .create(<NavLink to="/" />)
      .toJSON();
    expect(link).toMatchSnapshot();
    const selected = renderer
      .create(<NavLink to="/" selected />)
      .toJSON();
    expect(selected).toMatchSnapshot();
    const disabled = renderer
      .create(<NavLink to="/" disabled />)
      .toJSON();
    expect(disabled).toMatchSnapshot();
  });
});

describe("Mobile Nav Link", () => {
  it("renders correctly", () => {
    const link = renderer
      .create(<MobileNavLink to="/" />)
      .toJSON();
    expect(link).toMatchSnapshot();
    const selected = renderer
      .create(<MobileNavLink to="/" selected />)
      .toJSON();
    expect(selected).toMatchSnapshot();
    const disabled = renderer
      .create(<MobileNavLink to="/" disabled />)
      .toJSON();
    expect(disabled).toMatchSnapshot();
    const nested = renderer
      .create(<MobileNavLink to="/" nested />)
      .toJSON();
    expect(nested).toMatchSnapshot();
  });
});
