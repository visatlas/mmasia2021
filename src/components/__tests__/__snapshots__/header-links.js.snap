// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mobile Nav Link renders correctly 1`] = `
<a
  className="flex item-center w-full"
  href="/"
  title="Link"
>
  <span
    className="pl-8 pr-4 py-2 w-full text-white  hover:bg-menuHover block rounded-md 
  text-base font-semibold font-headingStyle"
  />
</a>
`;

exports[`Mobile Nav Link renders correctly 2`] = `
<a
  className="flex item-center w-full"
  href="/"
  title="Link"
>
  <span
    className="pl-8 pr-4 py-2 w-full text-white bg-menuSelected hover:bg-menuHover block rounded-md 
  text-base font-semibold font-headingStyle"
  />
</a>
`;

exports[`Mobile Nav Link renders correctly 3`] = `
<span
  className="pl-8 pr-4 py-2 flex item-center w-full text-textDisabled block rounded-md text-base  font-semibold font-headingStyle cursor-default"
/>
`;

exports[`Mobile Nav Link renders correctly 4`] = `
<a
  className="flex item-center w-full"
  href="/"
  title="Link"
>
  <span
    className="pl-16 pr-4 py-2 w-full text-white  hover:bg-menuHover block rounded-md 
  text-base font-medium font-headingStyle"
  />
</a>
`;

exports[`Nav Link renders correctly 1`] = `
<a
  className="flex item-center"
  href="/"
  title="Link"
>
  <span
    className=" text-white text-base hover:bg-menuHover px-3 py-2 rounded-md text-sm 
  font-medium font-headingStyle"
  />
</a>
`;

exports[`Nav Link renders correctly 2`] = `
<a
  className="flex item-center"
  href="/"
  title="Link"
>
  <span
    className="bg-menuSelected text-white text-base hover:bg-menuHover px-3 py-2 rounded-md text-sm 
  font-medium font-headingStyle"
  />
</a>
`;

exports[`Nav Link renders correctly 3`] = `
<span
  className="text-textDisabled text-base px-3 py-2 rounded-md text-sm font-medium  font-headingStyle cursor-default"
/>
`;
